### PR TITLE
bt-148: Add socket store middleware

### LINKED ISSUE #148: Add socket store middleware
- [x] Add socket package
- [x] Add socket store middleware
- [ ] Add chat bundle
- [x] Add mock for chat store (mock all actions that need)
- [x] Add join chat event
- [x] Add leave chat event
- [x] Add create message event

### PR DESCRIPTION
**Pull request by nika-sirak**
Added socket store middleware.

### CODE REVIEW

**Review by MaksymGavatuk — APPROVED** [review]
Looks good, but remove comment

File: frontend/src/bundles/chat/store/actions.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    //     receiverId: 'nskndkaldn',
+    //     senderId: 'nskndkaldn',
+    //     chatId: 'ksdjlskfjl',
+    // } = payload;
+    // const newMessage = await chatApi.createChatMessage({
+    //     message,
+    //     receiverId,
+    //     senderId,
+    //     chatId,
+    //   });
```
Comment (nikita-remeslov) [inline frontend/src/bundles/chat/store/actions.ts]: do we need this code?

File: frontend/src/bundles/chat/store/slice.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import { type ChatMessageGetAllItemResponseDto } from '../types/types.js';
+import { createMessage } from './actions.js';
+
+type State = {
+    currentChatMessages: ChatMessageGetAllItemResponseDto[];
+    currentChatId: string | null;
+    dataStatus: ValueOf<typeof DataStatus>;
+};
+
+const initialState: State = {
```
Comment (nikita-remeslov) [inline frontend/src/bundles/chat/store/slice.ts:15]: we should add info about other chats, to handle sidebar

File: frontend/src/bundles/chat/store/slice.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+};
+
+const { reducer, actions, name } = createSlice({
+    initialState,
+    name: 'chat',
+    reducers: {},
+    extraReducers(builder) {
+        builder
+            .addMatcher(isAnyOf(createMessage.pending), (state) => {
+                state.dataStatus = DataStatus.PENDING;
```
Comment (nikita-remeslov) [inline frontend/src/bundles/chat/store/slice.ts:28]: we can add logic to update store

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/chat/store/slice.ts (hunk 2 above)
Comment (nika-sirak) [reply frontend/src/bundles/chat/store/slice.ts:15]: I think it's related to #497

File: frontend/src/bundles/chat/store/slice.ts (hunk 3 above)
Comment (nika-sirak) [reply frontend/src/bundles/chat/store/slice.ts:28]: I think it's related to #497

File: frontend/src/bundles/chat/store/actions.ts (hunk 1 above)
Comment (nika-sirak) [reply frontend/src/bundles/chat/store/actions.ts]: Removed it.

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**